### PR TITLE
Add Cloudflare token auth support and fix Route53 imports

### PR DESCRIPTION
**Pull request by moritz89**
Thank you for contributing to sewer.                    
Every contribution to sewer is important to us.                       
You may not know it, but you have just contributed to making the world a more safer and secure place.

Contributor offers to license certain software (a “Contribution” or multiple
“Contributions”) to sewer, and sewer agrees to accept said Contributions,
under the terms of the MIT License.
Contributor understands and agrees that sewer shall have the irrevocable and perpetual right to make
and distribute copies of any Contribution, as well as to create and distribute collective works and
derivative works of any Contribution, under the MIT License.

Now,                   

## What(What have you changed?)
- Added support for tokens in Cloudflare DNS
- Fixed how Route53 imports dependencies

## Why(Why did you change it?)
- Tokens allow limiting the API functionality to the token
  - API keys allow access to all functionality
- Route53 throws import errors even when not used
- Fixes: https://github.com/komuw/sewer/issues/127

### CODE REVIEW

**Comment by komuw** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- sewer/dns_providers/tests/test_cloudflare.py  3
- sewer/cli.py  5
- sewer/dns_providers/cloudflare.py  3
         

Clones added
============
- sewer/dns_providers/tests/test_cloudflare.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/komuW/sewer/pullRequest?prid=4623330&bid=15502494)

**Comment by moritz89** [comment]
Closes #127

File: sewer/dns_providers/cloudflare.py [hunk 1]
```diff
@@ -149,3 +161,10 @@ def json(self):
                 )
 
         self.logger.info("delete_dns_record_success")
+
+    def _get_auth_header(self):
+        if self.CLOUDFLARE_TOKEN:
+            print(self.CLOUDFLARE_TOKEN)
```
Comment (komuw) [inline sewer/dns_providers/cloudflare.py]: @moritz89 print forgotten after debugging?

File: sewer/dns_providers/route53.py [hunk 2]
```diff
@@ -1,6 +1,11 @@
 import collections
-import boto3
-from botocore.client import Config
+
+try:
+    route53_dependencies = True
+    import boto3
+    from botocore.client import Config
+except ImportError:
+    route53_dependencies = False
```
Comment (komuw) [inline sewer/dns_providers/route53.py]: :+1:

**Comment by komuw** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- sewer/cli.py  5
- sewer/dns_providers/cloudflare.py  3
- sewer/dns_providers/tests/test_cloudflare.py  3
         

Clones added
============
- sewer/dns_providers/tests/test_cloudflare.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/komuW/sewer/pullRequest?prid=4623330&bid=15502494)

File: sewer/dns_providers/cloudflare.py (hunk 1 above)
Comment (moritz89) [reply sewer/dns_providers/cloudflare.py]: Haha, thanks for catching that

**Pull request merged**